### PR TITLE
Feature - optional number row

### PR DESCRIPTION
**Pull request by mohamediburan**
**Note:** This is not a complete PR. I just raised pull request to get some ideas on how to properly implement this feature as i want to contribute to my new fav keyboard.

Anyway, I made some progress on implementing the number row on QWERTY, QWERTZ and AZERTY keyboard. This is a naive approach as i still completely understand the code base. So, We can see the number row and it seems to be working fine for most part.

However, I am facing one major issue. When trying to switch to symbol keyboard from alphabet keyboard, it switches to symbol keyboard but quickly switches back to alphabet keyboard. I have traced the issue back to difference in height between the alphabet keyboard and symbol keyboard somehow causes it to click keys on following order. 
**Symbol key -> Shift key -> Symbol key** in milliseconds.

If i add some dummy keys to the symbol keyboard to match the height of the number row keyboard, then it works fine.

@dslul Could you please take a look at this and provide some suggestion on how to go about this issue and general code suggestions on how to improve?

### CODE REVIEW

File: app/src/main/java/org/dslul/openboard/inputmethod/latin/LatinIME.java [hunk 1]
```diff
@@ -1958,6 +1958,10 @@ public boolean shouldSwitchToOtherInputMethods() {
         return mRichImm.shouldOfferSwitchingToNextInputMethod(token, fallbackValue);
     }
 
+    public boolean shouldShowNumberRowKey() {
+        return mSettings.getCurrent().mShowsNumberRowKey;
+    }
+
```
Comment (dslul) [inline app/src/main/java/org/dslul/openboard/inputmethod/latin/LatinIME.java:1964]: This is not strictly necessary, it's better if you get the pref from KeyboardSwitcher directly

**Comment by dslul** [comment]
> When trying to switch to symbol keyboard from alphabet keyboard, it switches to symbol keyboard but quickly switches back to alphabet keyboard. I have traced the issue back to difference in height between the alphabet keyboard and symbol keyboard somehow causes it to click keys on following order.
> **Symbol key -> Shift key -> Symbol key** in milliseconds.
> 
> If i add some dummy keys to the symbol keyboard to match the height of the number row keyboard, then it works fine.
> 
> @dslul Could you please take a look at this and provide some suggestion on how to go about this issue and general code suggestions on how to improve?

Thanks for the contribution. This could be solved the "gboard way", by keeping the keyboard height fixed, and spacing more the symbol keys. I'm not sure if I like this behaviour, but I can't think of other solutions at the moment.

File: app/src/main/res/xml/rowkeys_number.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+<merge
+    xmlns:latin="http://schemas.android.com/apk/res-auto"
+    >
+    <!-- 1,2,3,4,5 -->
+    <include
+        latin:keyboardLayout="@xml/rowkeys_number_left5" />
+    <!-- 6,7,8,9,0 -->
+    <include
+        latin:keyboardLayout="@xml/rowkeys_number_right5" />
```
Comment (dslul) [inline app/src/main/res/xml/rowkeys_number.xml:29]: Why did you split the number row in two?

**Comment by Pentaphon** [comment]
> Thanks for the contribution. This could be solved the "gboard way", by keeping the keyboard height fixed, and spacing more the symbol keys. I'm not sure if I like this behaviour, but I can't think of other solutions at the moment.

I would go with that way until somebody thinks of something better. Lots of people want this number row when I recommend your keyboard to them.

**Comment by mohamediburan** [comment]
@dslul I checked "gboard" and i like it tbh. It's a very simple solution to this issue. Let me try.

**Comment by macbeto** [comment]
Another user here that will love to have that numbers row implementation in the next release :-)

**Comment by dslul** [comment]
@mohamediburan I wanted to push the update as soon as possible, so I ended up implementing this feature like it's done in Indic Keyboard, with an additional symbols row. Maybe I prefer like it's done in GBoard (with a smaller number row), but for this release I'll keep it this way. I'll keep this open in case you still want to implement it that way :)

**Comment by mohamediburan** [comment]
@dslul I checked out indic keyboard. That was also good in my opinion. Sure, will try my luck to try to make it like gboard if ibam able to. But for now, we have the option for numbers row, that's all i care about. Thanks for implementing this.

**Pull request closed without merging**